modules/data-entry/src/main/frontend/src/questionnaire/Form.jsx

```diff
@@ -319,6 +323,7 @@ function Form (props) {
               entryType={data?.questionnaire?.title || "Form"}
               shouldGoBack={true}
               buttonClass={classes.titleButton}
+              removeWindowHandlers={removeWindowHandlers}
```

Comment:
I don't think we need to be so specific and add a property named `removeWindowHandlers` to the `DeleteButton`. Technically this is something (anything) that is supposed to happen once _deletion is successful_. So we can make use of the `onComplete` handler and get rid of all changes done to the `DeleteButton` component.
```suggestion
              onComplete={removeWindowHandlers}
```

Reply:
Implemented by commit `8d67b37` :heavy_check_mark: